azext_iot/operations/events3/_builders.py

```diff
@@ -0,0 +1,143 @@
+        source = uamqp.address.Source(endpoint)
+        receive_client = uamqp.ReceiveClientAsync(source, auth=auth, timeout=30000, debug=DEBUG)
+        try:
+            await receive_client.open_async()
+            message = uamqp.Message(application_properties={'name': path})
+
+            response = await receive_client.mgmt_request_async(
+                message,
+                b'READ',
+                op_type=b'com.microsoft:eventhub',
+                status_code_field=b'status-code',
+                description_fields=b'status-description',
+                timeout=30000
+            )
+            test = response.get_data()
+            return test
+        finally:
+            await receive_client.close_async()
+
+
+    async def _evaluate_redirect(self, endpoint):
+        source = uamqp.address.Source('amqps://{}/messages/events/$management'.format(endpoint))
+        receive_client = uamqp.ReceiveClientAsync(source, timeout=30000, prefetch=1, debug=DEBUG)
+
+        try:
+            await receive_client.open_async()
+            await receive_client.receive_message_batch_async(max_batch_size=1)
+        except uamqp.errors.LinkRedirect as redirect:
+            redirect = unicode_binary_map(parse_entity(redirect))
+            result = {}
+            result['events'] = {}
+            result['events']['endpoint'] = redirect['hostname']
+            result['events']['path'] = redirect['address'].replace('amqps://', '').split('/')[1]
+            result['events']['address'] = redirect['address']
+            return redirect, result
+        finally:
+            await receive_client.close_async()
+
+
+    async def _buildCentralEventHubTargetAsync(self, cmd, app_id, aad_token):
+        from azext_iot.common._azure import get_iot_central_tokens
+
+        tokens = get_iot_central_tokens(cmd, app_id, aad_token)
+        eventHubToken = tokens['eventhubSasToken']
+        hostnameWithoutPrefix = eventHubToken['hostname'].split("/")[2]
+        endpoint = hostnameWithoutPrefix
+        path = eventHubToken["entityPath"]
+        tokenExpiry = tokens['expiry']
+        auth = self._build_auth_container_from_token(endpoint, path, eventHubToken['sasToken'], tokenExpiry)
+        address = "amqps://{}/{}/$management".format(hostnameWithoutPrefix, eventHubToken["entityPath"])
+        meta_data = await self._query_meta_data(address, path, auth)
+        partition_count = meta_data[b'partition_count']
+        partition_ids = []
+        for i in range(int(partition_count)):
+            partition_ids.append(str(i))
+        partitions = partition_ids
+        auth = self._build_auth_container_from_token(endpoint, path, eventHubToken['sasToken'], tokenExpiry)
+
+        eventHubTarget = {
+            'endpoint': endpoint,
+            'path': path,
+            'auth': auth,
+            'partitions': partitions
+        }
+
+        return eventHubTarget
+
+
+    # TODO: This is both here and in _events
```

Comment:
If the TODO is resolved, lets remove the comment to not confuse.

Reply:
I have now moved this method to a new class, AmqpBuilder for use in both places.